Sources/Drain.swift

```diff
@@ -39,17 +39,14 @@ public final class Drain: DataRepresentable, Stream {
         self.init(for: buffer.data)
     }
 
-    public func close() -> Bool {
-        if closed {
-            return false
-        }
+    public func close() throws {
+        guard !closed else { throw ClosableError.alreadyClosed }
         closed = true
```

Comment:
Personally I don't like this one-liner style. Feels too cluttered and we don't have any other example of that style in `C7`.
I think it would be best as:
```
guard !closed else {
    throw ClosableError.alreadyClosed
}
```